Makefile

```diff
@@ -71,7 +71,7 @@ docker-push:
 # download controller-gen if necessary
 controller-gen:
 ifeq (, $(shell which controller-gen))
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0
```

Comment:
I believe the test failures were because of this https://go.dev/doc/go-get-install-deprecation

Reply:
I think we might want to unconditionally run `go install`, so that it updates the installed version to the correct version? but I didn't want to change too much of the code

Reply:
Another approach would be to use `go run`, which has the benefit that it won't conflict with a `controller-gen` binary that someone already has installed; for example, we can change the generate target on line 50 as follows:

```makefile
# Generate code
generate:
	go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0 \
		object:headerFile=./hack/boilerplate.go.txt paths="./..."
```

We'd also have to update `manifests` similarly. Maybe we can pin the version using a variable? something like this should work:

```makefile
CONTROLLER_GEN ?= sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0

# Generate manifests e.g. CRD, RBAC etc.
manifests:
	go run $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases

# Generate code
generate:
	go run $(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
```

Then we can also remove the controller-gen target.

This runs pretty quickly because Go caches the command in the module cache:

```console
09:43 jawnsy@pattern ~/projects/gke-autoneg-controller $ go clean -modcache

09:44 jawnsy@pattern ~/projects/gke-autoneg-controller $ time make manifests
go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0 "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
go: downloading sigs.k8s.io/controller-tools v0.2.0
go: downloading github.com/spf13/cobra v0.0.3
go: downloading sigs.k8s.io/yaml v1.1.0
go: downloading golang.org/x/tools v0.0.0-20190501045030-23463209683d
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/gobuffalo/flect v0.1.5
go: downloading k8s.io/apiextensions-apiserver v0.0.0-20190409022649-727a075fdec8
go: downloading k8s.io/apimachinery v0.0.0-20190404173353-6a84e37a896d
go: downloading k8s.io/api v0.0.0-20190409021203-6e4e0e4f393b
go: downloading github.com/mattn/go-colorable v0.1.2
go: downloading github.com/mattn/go-isatty v0.0.8
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/spf13/pflag v1.0.3
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading github.com/google/gofuzz v0.0.0-20170612174753-24818f796faf
go: downloading k8s.io/klog v0.2.0
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading golang.org/x/sys v0.0.0-20190429190828-d89cdac9e872
go: downloading golang.org/x/net v0.0.0-20190501004415-9ce7a6920f09
go: downloading golang.org/x/text v0.3.2

real    0m12.328s
user    0m13.381s
sys     0m2.599s

09:45 jawnsy@pattern ~/projects/gke-autoneg-controller $ time make manifests
go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0 "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases

real    0m1.231s
user    0m2.405s
sys     0m0.465s

09:45 jawnsy@pattern ~/projects/gke-autoneg-controller $ time make generate
go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0 object:headerFile=./hack/boilerplate.go.txt paths="./..."

real    0m1.723s
user    0m4.685s
sys     0m0.604s
```

Reply:
Sounds like `go run` would make the most sense.

Reply:
Updated to use `go run`, PTAL!